python/packages/ai/teams/ai/ai.py

```diff
@@ -76,3 +100,221 @@ def __call__(func: Callable[[TurnContext, StateT, Any, str], Awaitable[bool]]):
             return func
 
         return __call__
+
+    def review_prompt(
+        self, func: Callable[[TurnContext, StateT, PromptTemplate], Awaitable[Optional[Plan]]]
+    ):
+        """
+        Reviews user input before it's sent to the planner.
+
+        ```python
+        # Use this method as a decorator
+        @app.ai.review_prompt
+        async def on_review_prompt(context: TurnContext, state: TurnState, prompt: PromptTemplate):
+            return None
+
+        # Pass a function to this method
+        app.ai.review_prompt(on_review_prompt)
+        ```
+
+        Args:
+            context (TurnContext): The turn context.
+            state (StateT): The turn state.
+            prompt (PromptTemplate): The prompt.
+
+        Returns:
+            Optional[Plan]: Returns None if the moderator approves the input,
```

Comment:
Line character length seems kind of short to me. I'm inclined to recommend making it longer. `js` is 120, I believe.